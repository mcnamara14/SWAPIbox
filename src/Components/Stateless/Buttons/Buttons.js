import React, { Component } from 'react';
import './Buttons.css';

class Buttons extends Component {
  constructor(props){
    super(props)
  }

  render() {
    return (
      <section className="filterButtons">
        <button className="people" onClick={() => this.props.setData('people')} >people</button>
        <button className="planets" onClick={() => this.props.setData('planets')} >planets</button>
        <button className="vehicles" onClick={() => this.props.setData('vehicles')} >vehicles</button>
      </section>
    );
  }
}

export default Buttons;
