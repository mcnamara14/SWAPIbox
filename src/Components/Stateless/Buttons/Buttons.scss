.header {
  width: 80%;
  display: inline-block;
}

.filterButtons {
  background: #241c41;
  border-top: 1px solid #624385;
  height: 80px;
  border-bottom: 3px solid #fe6244;
}

.filterButtons div {
  width: 80%;
  margin: 0 auto;
  height: 100%;
}

button {
  background: none;
  border: none;
  color: #ccc;
  margin-top: -2px;
  text-transform: uppercase;
  font-size: 1rem;
  width: 25%;
  height: 99%;
  border-left: 1px solid #302752;
  border-right: 1px solid #302752;
  font-family: 'Exo 2', sans-serif;
  font-weight: 700;
  letter-spacing: 2px;
}

button.favorites {
  margin-top: 0px;
}

button:hover {
  cursor: pointer;
  background: #302752;
}

button:focus {
  outline: none;
}

.favorites span {
  color: white;
  margin-left: 10px;
  margin-bottom: 0px;
  width: 13px;
  font-size: 1.3rem;
  background: #302752;
  border: 1px solid #555;
  padding: 6px 12px 5px 9px;
  display: inline-block;
}

@media screen and (max-width: 1000px) {
  .filterButtons {
    height: 80px;
  }

  button {
    margin-top: 0px;
    font-size: 0.8rem;
  }

  .card {
    width: 100%;
    text-align: center;
    height: 450px;
  }

  .card p {
    margin: 6% 10%;
    font-size: 1.3rem;
  }

  .card span {
    font-size: 1.2rem;
  }

  h2 {
    font-size: 1.3rem;
  }

  .cardFave {
    width: 18% !important;
  }

  .favorites span {
    border: none;
    padding: 0;
    background: none;
    margin-left: 5px;
    font-size: 1.2rem
  }

  .filterButtons div {
    width: 100%;
    margin: 0 auto;
    height: 100%;
}

}