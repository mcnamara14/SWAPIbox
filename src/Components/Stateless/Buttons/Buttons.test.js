import React from 'react';
import ReactDOM from 'react-dom';
import Buttons from './Buttons';
import { shallow } from 'enzyme';

describe('Buttons', () => {
  let wrapper; 

  beforeEach(() => {
    wrapper = shallow(<Buttons favoriteCount={ 1 } setData={ jest.fn() } toggleDisplayFavorites={ jest.fn() } />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('handleClick', () => {

    it('should call toggleDisplayFavorites if favoriteCount is more than 0', () => {
      wrapper.instance().handleClick();

      expect(wrapper.instance().props.toggleDisplayFavorites).toHaveBeenCalled();
    })

    it('should not call toggleDisplayFavorites if favoriteCount is 0', () => {
      wrapper = shallow(<Buttons favoriteCount={ 0 } toggleDisplayFavorites={ jest.fn() }/>);

      wrapper.instance().handleClick();

      expect(wrapper.instance().props.toggleDisplayFavorites).not.toHaveBeenCalled();
    })
  })

  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Buttons setData={ jest.fn() } toggleDisplayFavorites={ jest.fn() } />);
    })

    it('should call handleClick when favorites is clicked', () => {
      wrapper.instance().handleClick = jest.fn();
      wrapper.find('.favorites').simulate('click');
  
      expect(wrapper.instance().handleClick).toHaveBeenCalled();
    });

    it('should call setData when people is clicked', () => {
      wrapper.find('.people').simulate('click');
  
      expect(wrapper.instance().props.setData).toHaveBeenCalled();
    });

    it('should call setData when planets is clicked', () => {
      wrapper.find('.planets').simulate('click');
  
      expect(wrapper.instance().props.setData).toHaveBeenCalled();
    });

    it('should call setData when vehicles is clicked', () => {
      wrapper.find('.vehicles').simulate('click');
  
      expect(wrapper.instance().props.setData).toHaveBeenCalled();
    });
  })
})

