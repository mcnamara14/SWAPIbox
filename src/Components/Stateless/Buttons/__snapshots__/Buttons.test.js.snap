// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Buttons
    favoriteCount={1}
    setData={[Function]}
    toggleAddFavorites={[Function]}
    toggleDisplayFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <button
                className="people"
                onClick={[Function]}
        >
                people
        </button>
        <button
                className="planets"
                onClick={[Function]}
        >
                planets
        </button>
        <button
                className="vehicles"
                onClick={[Function]}
        >
                vehicles
        </button>
        <button
                className="favorites"
                onClick={[Function]}
        >
                View Favorites
                <span>
                        1
                </span>
        </button>
</div>,
      "className": "filterButtons",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="people"
            onClick={[Function]}
>
            people
</button>,
          <button
            className="planets"
            onClick={[Function]}
>
            planets
</button>,
          <button
            className="vehicles"
            onClick={[Function]}
>
            vehicles
</button>,
          <button
            className="favorites"
            onClick={[Function]}
>
            View Favorites
            <span>
                        1
            </span>
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people",
            "className": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "people",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "planets",
            "className": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "vehicles",
            "className": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "View Favorites",
              <span>
                1
</span>,
            ],
            "className": "favorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "View Favorites",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <button
                    className="people"
                    onClick={[Function]}
          >
                    people
          </button>
          <button
                    className="planets"
                    onClick={[Function]}
          >
                    planets
          </button>
          <button
                    className="vehicles"
                    onClick={[Function]}
          >
                    vehicles
          </button>
          <button
                    className="favorites"
                    onClick={[Function]}
          >
                    View Favorites
                    <span>
                              1
                    </span>
          </button>
</div>,
        "className": "filterButtons",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="people"
              onClick={[Function]}
>
              people
</button>,
            <button
              className="planets"
              onClick={[Function]}
>
              planets
</button>,
            <button
              className="vehicles"
              onClick={[Function]}
>
              vehicles
</button>,
            <button
              className="favorites"
              onClick={[Function]}
>
              View Favorites
              <span>
                            1
              </span>
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "people",
              "className": "people",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "people",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "planets",
              "className": "planets",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "planets",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "vehicles",
              "className": "vehicles",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "vehicles",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "View Favorites",
                <span>
                  1
</span>,
              ],
              "className": "favorites",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "View Favorites",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "span",
              },
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
